jupyter_events/schema.py

```diff
@@ -55,23 +57,27 @@ class EventSchema:
         any schema registered here follows the expected form
         of Jupyter Events.
 
-    resolver:
-        RefResolver for nested JSON schema references.
+    registry:
+        Registry for nested JSON schema references.
     """
 
     def __init__(
         self,
         schema: SchemaType,
         validator_class: Type[Validator] = validators.Draft7Validator,  # type:ignore[assignment]
         format_checker: FormatChecker = draft7_format_checker,
-        resolver: Optional[RefResolver] = None,
+        registry: Optional[Registry] = None,
     ):
         """Initialize an event schema."""
         _schema = self._load_schema(schema)
         # Validate the schema against Jupyter Events metaschema.
         validate_schema(_schema)
+
+        if registry is None:
+            registry = Registry().with_resource(_schema["$id"], DRAFT7.create_resource(_schema))
```

Comment:
Hey! Just because I'm keeping an eye on how the referencing rollout is going (and love `jupyter`, thanks to you all!), I think a minor tidy here would be:

```suggestion
            registry = DRAFT7.create_resource(_schema) @ Registry()
```

or even `Resource.from_contents(schema) @ Registry()` if you didn't want to hard code that your schema is draft 7 -- that will use the value you set at `$schema` to decide what version it is.

But either way that should obviate you needing to bother with pulling out the `$id` yourself.

(Docs are [here](https://referencing.readthedocs.io/en/stable/api/#referencing.Registry.__rmatmul__), suggestions always welcome if there are improvements).

And yeah feel free to ping me or open issues if you encounter any!

Reply:
Thank you for the suggestion @Julian!
I pushed a new commit applying them.